llvm5/plan.sh

```diff
@@ -2,11 +2,11 @@ source "$(dirname "${BASH_SOURCE[0]}")/../llvm/plan.sh"
 
 pkg_name=llvm5
 pkg_origin=core
-pkg_version=5.0.1
+pkg_version=5.0.2
```

Comment:
i wonder if llvm5.0.2 can be built with gcc8

@TheDarkula can you please verify?

cc: @smacfarlane

Reply:
I did test 5.0.2 as part of the refresh to see if it would build and was unsuccessful.  I do think it would be good to double-check though 😄

Reply:
I just ran through a build and it did fail:

```
[2064/2506] Building CXX object tools/lli/CMakeFiles/lli.dir/lli.cpp.o
FAILED: tools/lli/CMakeFiles/lli.dir/lli.cpp.o 
/hab/pkgs/core/gcc/8.2.0/20190115004042/bin/c++  -DLLVM_BUILD_GLOBAL_ISEL -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Itools/lli -I/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli -Iinclude -I/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include -I/hab/pkgs/core/glibc/2.27/20190115002733/include -I/hab/pkgs/core/libffi/3.2.1/20190115154244/lib/libffi-3.2.1/include -I/hab/pkgs/core/python2/2.7.15/20190115155422/include -I/hab/pkgs/core/zlib/1.2.11/20190115003728/include -I/hab/pkgs/core/cmake/3.10.2/20190201010140/include -I/hab/pkgs/core/gcc/8.2.0/20190115004042/include -fPIC -fvisibility-inlines-hidden -Werror=date-time -std=c++11 -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -ffunction-sections -fdata-sections -O3 -DNDEBUG    -fno-exceptions -MD -MT tools/lli/CMakeFiles/lli.dir/lli.cpp.o -MF tools/lli/CMakeFiles/lli.dir/lli.cpp.o.d -o tools/lli/CMakeFiles/lli.dir/lli.cpp.o -c /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/lli.cpp
In file included from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/CodeGen/LinkAllCodegenComponents.h:20,
                 from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/lli.cpp:22:
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/CodeGen/SchedulerRegistry.h: In constructor 'llvm::RegisterScheduler::RegisterScheduler(const char*, const char*, llvm::RegisterScheduler::FunctionPassCtor)':
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/CodeGen/SchedulerRegistry.h:40:52: warning: cast between incompatible function types from 'llvm::RegisterScheduler::FunctionPassCtor' {aka 'llvm::ScheduleDAGSDNodes* (*)(llvm::SelectionDAGISel*, llvm::CodeGenOpt::Level)'} to 'llvm::MachinePassCtor' {aka 'void* (*)()'} [-Wcast-function-type]
   : MachinePassRegistryNode(N, D, (MachinePassCtor)C)
                                                    ^
In file included from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/lli.cpp:30:
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/ExecutionEngine/Orc/OrcRemoteTargetClient.h: In member function 'llvm::Expected<std::vector<char> > llvm::orc::remote::OrcRemoteTargetClient<ChannelT>::readMem(char*, llvm::JITTargetAddress, uint64_t)':
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/ExecutionEngine/Orc/OrcRemoteTargetClient.h:722:26: error: could not convert '((llvm::orc::remote::OrcRemoteTargetClient<ChannelT>*)this)->callB<llvm::orc::remote::OrcRemoteTargetRPCAPI::ReadMem>(Src, Size)' from 'Expected<vector<unsigned char,allocator<unsigned char>>>' to 'Expected<vector<char,allocator<char>>>'
     return callB<ReadMem>(Src, Size);
            ~~~~~~~~~~~~~~^~~~~~~~~~~
In file included from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/ADT/STLExtras.h:32,
                 from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/OrcLazyJIT.h:18,
                 from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/lli.cpp:16:
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/ADT/SmallVector.h: In instantiation of 'void llvm::SmallVectorTemplateBase<T, true>::push_back(const T&) [with T = std::pair<void*, long unsigned int>]':
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/Support/Allocator.h:247:33:   required from 'void* llvm::BumpPtrAllocatorImpl<AllocatorT, SlabSize, SizeThreshold>::Allocate(size_t, size_t) [with AllocatorT = llvm::MallocAllocator; long unsigned int SlabSize = 4096; long unsigned int SizeThreshold = 4096; size_t = long unsigned int]'
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/Support/YAMLParser.h:130:42:   required from here
/hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/ADT/SmallVector.h:309:11: warning: 'void* memcpy(void*, const void*, size_t)' writing to an object of type 'struct std::pair<void*, long unsigned int>' with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
     memcpy(this->end(), &Elt, sizeof(T));
     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /hab/pkgs/core/gcc/8.2.0/20190115004042/include/c++/8.2.0/utility:70,
                 from /hab/pkgs/core/gcc/8.2.0/20190115004042/include/c++/8.2.0/algorithm:60,
                 from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/include/llvm/ADT/STLExtras.h:20,
                 from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/OrcLazyJIT.h:18,
                 from /hab/pkgs/chainetix/llvm5/5.0.2/20190204180643/src/tools/lli/lli.cpp:16:
/hab/pkgs/core/gcc/8.2.0/20190115004042/include/c++/8.2.0/bits/stl_pair.h:198:12: note: 'struct std::pair<void*, long unsigned int>' declared here
     struct pair
            ^~~~
[2066/2506] Building CXX object tools/lli/CMakeFiles/lli.dir/OrcLazyJIT.cpp.o
ninja: build stopped: subcommand failed.
   llvm5: Build time: 37m41s
   llvm5: Exiting on error
```

Reply:
Thanks for confirming!  Could you revert the llvm5 changes, and then this can go in after #2239

Reply:
👍 thanks for checking!